projects/extension/src/background/ConnectionManager.ts

```diff
@@ -153,6 +156,23 @@ export class ConnectionManager extends (EventEmitter as {
     return client.terminate()
   }
 
+  flushDatabases(): void {
```

Comment:
`flushDatabases` cannot be outside the `ConnectionManager` as it needs access to the private `#chainConnections` (more specific to the `chain` that is not exposed). Based on @tomaka 's comment in Element `we should pass the database content in each instance of the known chains`

Reply:
In order to receive the `databaseContent` content you need access to the `chain` after it is connected. That is possible only inside the `ConnectionManager` as the `chain` is part of the private `#chainConnections` (I dont think its a good idea to export it for security issues).

Reply:
that's inaccurate, `manager.addChain` which is already used from the `init` function that I mentioned, already returns a promise with the `chain`. So, the init function already is receiving the `chain` is just that until now it's not using it, bu it could.